packages/jest-emotion/test/react-enzyme.test.js

```diff
@@ -1,130 +1,216 @@
-import 'test-utils/legacy-env'
+import 'test-utils/legacy-env' // adds serializer
 /** @jsx jsx */
+import jestInCase from 'jest-in-case'
 import * as enzyme from 'enzyme'
-import { jsx } from '@emotion/core'
-import { createSerializer as createEnzymeSerializer } from 'enzyme-to-json'
-import { createSerializer } from 'jest-emotion'
-import { toMatchSnapshot } from 'jest-snapshot'
+import { jsx, ThemeProvider } from '@emotion/core'
+import styled from '@emotion/styled'
 import React from 'react'
+import toJson from 'enzyme-to-json'
 
-const createEnzymeSnapshotMatcher = serializerOptions => {
-  const serializer = createEnzymeSerializer(serializerOptions)
-  const identityPrinter = v => v
+afterEach(() => {
```

Comment:
maybe just smth like `document.head.innerHTML = ''`?

Reply:
That didn't seem to work, but can take a look after the caching issue is solved. As it stands, clearing the DOM breaks subsequent tests if they use the same styles, since emotion believes the styles have already been added to the DOM as they exist in emotion's cache.

Reply:
you don't clear the cache in this afterEach though, it's only about clearing the emotion-generated style tags, so a simple document.head overwrite should work, right? it's just nitpicking, your version actually covers more cases

Reply:
And not sure if I've commented about this anywhere - but "clearing" the cache could be accomplished by wrapping each test in a separate CacheProvider